src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -41,6 +42,21 @@
 #define TRANSIENT_FAILURE_ATTEMPTS 3 /* Arbitrary limit: how many attempts are considered enough to report
                                       * transient failure. */
 
+/* RFC2131 section 4.1 specifies retransmission delays, starting at 4s then doubling
+ * each attempt with max of 64s, with -1 to +1 sec of random 'fuzz' added. */
+#define RFC2131_APPLY_FUZZ(time) \
+        usec_sub_signed((time), ((int64_t)(random_u64() % (2 * USEC_PER_SEC)) - (int64_t)USEC_PER_SEC))
+#define RFC2131_SEC41_NEXT_TIMEOUT(now, attempt) \
+        RFC2131_APPLY_FUZZ(usec_add((now), ((UINT64_C(1) << MIN((attempt) + 1, UINT64_C(6))) * USEC_PER_SEC)))
+
+/* RFC2131 section 4.4.5 specifies T1 (renewing) and T2 (rebinding) defaults and
+ * special reacquisition delay algorithm, different from above section 4.1,
+ * however we use the same -1 to +1 sec 'fuzz' from above */
+#define RFC2131_T1_DEFAULT(lifetime) ((lifetime) / 2)
+#define RFC2131_T2_DEFAULT(lifetime) (((lifetime) * 7) / 8)
+#define RFC2131_SEC445_NEXT_TIMEOUT(now, expire) \
+        RFC2131_APPLY_FUZZ(usec_add((now), MAX(usec_sub_unsigned((expire), (now)) / 2, (60 * USEC_PER_SEC))))
```

Comment:
`now` is used twice.

Reply:
ok, i changed the macros to be functions instead; i think it's a bit clearer and avoids the double-evaluation of now.